pkg/queue/doc.go

```diff
@@ -0,0 +1,6 @@
+// Copyright 2022 Outreach Corporation. All Rights Reserved.
+//
+// Description: queue data struct.
+
+// Package queue provides queue data struct.
+package queue
```

Comment:
This feels like the wrong package name

Reply:
If the reviewers here are ok with it, but when I think queue I would expect a standard queue implementation first.  Maybe this should be data_structures or whatever is go idiomatic?

Reply:
There could be other queue implementation with any/combination of the following characteristics, they all fit in the queue package.
* Blocking: block on push/pop
* FIFO: first in first out
* TTL: item will TTL
* Stream: in memory kafa
* Delay: item in queue is available with a delay

Reply:
I understand different queue types, I am just saying right now we do not have those and it is weird to come to a queue package that has one queue that is a more esoteric queue type.  All of those are likely backed by a real external queue system and the implementation here would just be a wrapper and likely not right for gobox.

Reply:
Any naming recommendation? Or moving this package to `queue/priorityqueue`?

Reply:
Will use `queue` for now, unless another package name is proposed.